enumerables.rb

```diff
@@ -1,37 +1,72 @@
 module Enumerables
+
   def my_each
-    count = 0
-    while count < self.size do
-      yield(self[count])
-      count += 1
+    return to_enum unless block_given?
+
+    for i in 0..self.length
+      yield(self[i])
     end
-    self
   end
 
   def my_each_with_index
-    
+    return to_enum unless block_given?
+    my_index = 0
+    self.my_each do |i|
+      yield i, my_index
+      my_index += 1
+    end
   end
 
   def my_select
     my_arr = []
-    count = 0
-    while count < self .length
-      my_arr << (self[count]) if yield(self[count]) == true
-      count += 1
-    end
+    self.my_each{ |x| my_arr << x if yield(x) }
     my_arr
   end
 
-  def my_all?
-  end
+    def my_all?
+      self.my_each do |index|
+        return true unless block_given?
+        true_false = yield(index)
+        return false unless true_false
+      end
+      true
+      to_enum
+    end
 
-  def my_any?
+  def my_any?(elem = 0)
+    unless block_given?
+      if elem.instance_of?(Class)
+        my_each { |x| return true if x.is_a? elem }
+      elsif elem.instance_of?(Regexp)
+        my_each { |x| return true if elem.match?(x.to_s) }
+      elsif [nil, false].include?(elem)
+        my_each { |x| return true if x == elem }
+      end
+      return false
+    end
+    my_each { |x| return true if yield(x) }
+    false
   end
 
   def my_none?
+    my_each do |y|
+      return false if block_given? && yield(y) || !block_given? && y
+    end
+    true
+    to_enum
   end
```

Comment:
- [ ] enumerables #my_none? returns true if the block never returns true for all elements
      Failure/Error: let(:true_block) { proc { |num| num > HIGHEST_VALUE } }

- [ ] enumerables #my_none? returns false if the block ever returns true for all elements::range
      Failure/Error: expect(range.my_none?(&false_block)).to eq(range.none?(&false_block))

- [ ]  enumerables #my_none? when no block or argument is given returns true only if none of the collection members is true
      Failure/Error: expect(false_array.my_none?).to be true
      
        expected true
             got #<Enumerator:1300> => #<Enumerator: [nil, false, nil, false]:each>

- [ ] enumerables #my_none? when a class is passed as an argument returns true if none of the collection is a member of such class
      Failure/Error: expect(array.my_none?(String)).to be array.my_none?(String)
      
      ArgumentError:
        wrong number of arguments (given 1, expected 0)

- [ ] enumerables #my_none? when a class is passed as an argument returns true if none of the collection is a member of such class::Numeric
      Failure/Error: expect(array.my_none?(Numeric)).to be array.my_none?(Numeric)
      
      ArgumentError:
        wrong number of arguments (given 1, expected 0)

- [ ]  enumerables #my_none? when a class is passed as an argument returns false if any of the collection is a member of such class
      Failure/Error: expect(array.my_none?(String)).to be array.my_none?(String)
      
      ArgumentError:
        wrong number of arguments (given 1, expected 0)

- [ ]  enumerables #my_none? when a Regex is passed as an argument returns true if none of the collection matches the Regex
      Failure/Error: expect(words.my_none?(/z/)).to be words.none?(/z/)
      
      ArgumentError:
        wrong number of arguments (given 1, expected 0)

- [ ]  enumerables #my_none? when a Regex is passed as an argument returns false if any of the collection matches the Regex
      Failure/Error: expect(words.my_none?(/d/)).to be words.none?(/d/)
      
      ArgumentError:
        wrong number of arguments (given 1, expected 0)

- [ ] enumerables #my_none? when a pattern other than Regex or a Class is given returns true only if none of the collection matches the pattern
      Failure/Error: expect(words.my_none?(5)).to be words.none?(5)
      
      ArgumentError:
        wrong number of arguments (given 1, expected 0)

- [ ]  enumerables #my_none? when a pattern other than Regex or a Class is given returns false only if one of the collection matches the pattern
      Failure/Error: expect(words.my_none?(5)).to be words.none?(5)
      
      ArgumentError:
        wrong number of arguments (given 1, expected 0)